File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.py
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        maxCandy = max(candies)
        result = []
        for each in candies:
            if each + extraCandies >= maxCandy :
                result.append(True)
            else:
                result.append(False)
        
        return result

File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        int maximumCandies = Arrays.stream(candies).max().getAsInt();
        List<Boolean> result = new ArrayList<>();
        
        for (int eachCandy: candies){
            if(eachCandy + extraCandies >= maximumCandies) {
                result.add(true);
            } else{
                result.add(false);
            }
        }
        
        return result;
    }
}